rw.go

```diff
@@ -42,10 +42,10 @@ func (w *etmWriter) WriteMsg(b []byte) error {
 	w.Lock()
 	defer w.Unlock()
 
-	bufsize := uint32(4 + len(b) + w.mac.Size())
 	// encrypt.
-	buf := mpool.ByteSlicePool.Get(bufsize).([]byte)
-	data := buf[4 : 4+len(b)] // the pool's buffer may be larger
```

Comment:
This comment was wrong. We have to do this because we over-allocate to have space for the mac